tensorflow/core/util/gpu_kernel_helper.h

```diff
@@ -30,8 +30,17 @@ limitations under the License.
 #if GOOGLE_CUDA
 #define TF_RED_WARPSIZE 32
 #elif TENSORFLOW_USE_ROCM
-// We don't define TF_RED_WARPSIZE here, because it can be either 32 or 64
-// and the value is not known at compile time.
+
+inline int AMD_WarpSize() {
+  hipDeviceProp_t props;
+  int dev = 0;
+  hipError_t result = hipGetDevice(&dev);
+  result = hipGetDeviceProperties(&props, dev);
+  if (result == hipSuccess) {
+    return props.warpSize;
+  }
+  return 64;  // AMD default warp size for CDNA
+}
```

Comment:
Is this code overwriting `result`? If yes, why?

Perhaps something like this is more clear?
```c++
inline int AMD_WarpSize() {
    hipDeviceProp_t props;
    int dev = 0;

    if (hipGetDevice(&dev) != hipSuccess ||
        hipGetDeviceProperties(&props, dev) != hipSuccess) {
        return 64;
    }

    return props.warpSize;
}
```

Reply:
Actually, I think this is better

```
inline hipError_t AMD_WarpSize(int* warpSize) {
  hipDeviceProp_t props;
  int dev = 0;

  hipError_t result = hipGetDevice(&dev);
  if (result != hipSuccess) {
    return result;
  }
  result = hipGetDeviceProperties(&props, dev);
  if (result != hipSuccess) {
    return result;
  }
  *warpSize = props.warpSize;
  return hipSuccess;
}
```

Reply:
Okay yeah if you'd rather it handle the error types, this makes more sense.